### ISSUE TITLE
Share Terminal interface between core and addons

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#1221]
Both core and the addons should be able to point at the .d.ts file that holds our public API. We can also base our internal ITerminal interface off the `Terminal` class declaration in xterm.d.ts.

This is blocked on https://github.com/xtermjs/xterm.js/issues/1220

**Comment by Tyriar** [comment]
Already done
